wpengine/atlas-content-modeler
# MTKA-1534: Add wp acm reset WP-CLI command

## Description

Adds a `wp acm reset` WP-CLI command to remove ACM models, taxonomies, taxonomy terms, posts, relationship data and media items.

Intended for use for blueprint authors to reset content to a clean state.

Pass `--all` to also remove core posts and related media.

Pass `--yes` to skip the confirmation prompt.

https://wpengine.atlassian.net/browse/MTKA-1534

## Checklist

I have:


## Testing

Includes new unit tests.

To test manually:

1. Import blueprint content.

```
> wp acm blueprint import https://raw.githubusercontent.com/wpengine/atlas-blueprint-portfolio/main/acm-blueprint.zip
Cannot load Zend OPcache - it was already loaded
Fetching blueprint.
Unzipping.
Verifying ACM manifest.
Checking minimum versions.
Importing ACM models and fields.
Importing posts.
Importing terms.
Tagging posts.
Importing media.
Importing post meta.
Restoring WordPress options.
Deleting zip and manifest.
Success: Import complete.
```

2. Remove blueprint content relating to ACM post types:

```
> wp acm reset --yes
Cannot load Zend OPcache - it was already loaded
Deleting taxonomy terms.
Deleting ACM taxonomies.
Deleting relationship data.
Deleting posts.
Deleting media.
Deleting models.
Deleted: 17 posts, 12 media, 2 models, 0 terms, 0 taxonomies, 0 relationships.
Success: ACM reset complete.
```

3. Pass `--all` to also remove content relating to core posts:

```
> wp acm reset --yes --all
Cannot load Zend OPcache - it was already loaded
Deleting taxonomy terms.
Deleting ACM taxonomies.
Deleting relationship data.
Deleting posts.
Deleting media.
Deleting models.
Deleted: 31 posts, 26 media, 0 terms, 0 taxonomies, 0 relationships, 0 models.
Success: ACM reset complete.
```

4. There is now nothing left to remove, so subsequent runs should report that nothing was deleted:

```
> wp acm reset --yes --all
Cannot load Zend OPcache - it was already loaded
Deleting taxonomy terms.
Deleting ACM taxonomies.
Deleting relationship data.
Deleting posts.
Deleting media.
Deleting models.
Deleted: 0 terms, 0 taxonomies, 0 relationships, 0 posts, 0 media, 0 models.
Success: ACM reset complete.
```

Running `wp acm reset --help` should output help info:

```
NAME

  wp acm reset

DESCRIPTION

  Resets ACM by deleting models, taxonomies, taxonomy terms, posts, relationship data and media items relating to ACM models.

SYNOPSIS

  wp acm reset [--yes] [--all]

OPTIONS

  [--yes]
    Skip prompt to confirm deletion.

  [--all]
    Delete all published posts, pages, custom posts and media, not just posts and media associated with ACM models.

EXAMPLES

    wp acm reset
    wp acm reset --yes
    wp acm reset --all
    wp acm reset --yes --all
```

## Documentation Changes

Documented the new reset command at the bottom of this page:

https://github.com/wpengine/atlas-content-modeler/blob/mtka-1534/wp-acm-reset/docs/wp-cli/index.md
